docs/configuration.md

```diff
@@ -3134,3 +3134,119 @@ The stage level scheduling feature allows users to specify task and executor res
+  </td>
+  <td>3.2.0</td>
+</tr>
+<tr>
+  <td><code>spark.shuffle.server.minChunkSizeInMergedShuffleFile</code></td>
+  <td><code>2m</code></td>
+  <td>
+    The minimum size of a chunk when dividing a merged shuffle file into multiple chunks during push-based shuffle. A merged shuffle file consists of multiple small shuffle blocks. Fetching the complete merged shuffle file in a single disk I/O increases the memory requirements for both the clients and the shuffle service. Instead, the shuffle service serves the merged file in <code> MB-sized chunks </code>. This configuration controls how big a chunk can get. A corresponding index file for each merged shuffle file will be generated indicating chunk boundaries.
+  </td>
+  <td>3.2.0</td>
+</tr>
+<tr>
+  <td><code>spark.shuffle.server.mergedIndexCacheSize</code></td>
+  <td><code>100m</code></td>
+  <td>
+    The size of cache in memory which is used in push-based shuffle for storing merged index files. This cache is in addition to the one configured via <code>spark.shuffle.service.index.cache.size</code>.
+  </td>
+  <td>3.2.0</td>
+</tr>
+<tr>
+  <td><code>spark.shuffle.server.ioExceptionsThresholdDuringMerge</code></td>
+  <td><code>4</code></td>
+  <td>
+    The threshold for number of IOExceptions while merging shuffle blocks to a shuffle partition. When the number of IOExceptions while writing to the merged shuffle data/index/meta files exceed this threshold then the shuffle server will stop merging shuffle  blocks for this shuffle partition.
+  </td>
+  <td>3.2.0</td>
+</tr>
+</table>
+
+### Client side configuration options
+
+<table class="table">
+<tr><th>Property Name</th><th>Default</th><th>Meaning</th><th>Since Version</th></tr>
+<tr>
+  <td><code>spark.shuffle.push.enabled</code></td>
+  <td><code>false</code></td>
+  <td>
+    Set to 'true' to enable push-based shuffle on the client side and this works in conjunction with the server side flag spark.shuffle.server.mergedShuffleFileManagerImpl which needs to be set with the appropriate org.apache.spark.network.shuffle.MergedShuffleFileManager implementation for push-based shuffle to be enabled
+  </td>
+  <td>3.2.0</td>
+</tr>
+<tr>
+  <td><code>spark.shuffle.push.merge.finalize.timeout</code></td>
+  <td><code>10s</code></td>
+  <td>
+    Specify the amount of time DAGScheduler waits after all mappers finish for a given shuffle map stage before it starts sending merge finalize requests to remote shuffle services. This allows the shuffle services some extra time to merge as many blocks as possible.
+  </td>
+ <td>3.2.0</td>
+</tr>
+<tr>
+  <td><code>spark.shuffle.push.maxRetainedMergerLocations</code></td>
+  <td><code>500</code></td>
+  <td>
+    Maximum number of shuffle push merger locations cached for push based shuffle. Currently, shuffle push merger locations are nothing but external shuffle services which are responsible for handling pushed blocks and merging them and serving merged blocks for later shuffle fetch.
```

Comment:
`...but the host of external shuffle services`?

Reply:
`Maximum number of shuffle push merger locations cached for push based shuffle. Currently, shuffle push merger locations are nothing but external shuffle services which are responsible for handling pushed blocks and merging them and serving merged blocks for later shuffle fetch.` ->


`Maximum number of merger locations cached for push based shuffle. Currently, merger locations are hosts of external shuffle services responsible for handling pushed blocks, merging them and serving merged blocks for later shuffle fetch.`